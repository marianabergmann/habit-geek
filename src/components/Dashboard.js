import React from "react";
import "./Dashboard.scss";

const Dashboard = ({ name, habits }) => {
  const habitComponents = habits.map((habit, index) => (
    <div className="dashboard__row" key={`habit-${index}`}>
      {habit}
    </div>
  ));

  const checkComponents = habits.map((habit, containerIndex) => (
    <div key={`check-container-${containerIndex}`} className="dashboard__row">
      {[...Array(7).keys()].map(checkIndex => (
        <input
          key={`check-${containerIndex}-${checkIndex}`}
          className="dashboard__cell"
          type="checkbox"
        ></input>
      ))}
    </div>
  ));

  return (
    <div className="dashboard">
      <h2>Do your best{name ? " " + name : ""}!</h2>
      <div className="dashboard__container">
        <div className="dashboard__habits">
          <div className="dashboard__row"></div>
          {habitComponents}
        </div>
        <div className="dashboard__days">
          <div className="dashboard__row">
            <div className="dashboard__cell">Mon</div>
            <div className="dashboard__cell">Tue</div>
            <div className="dashboard__cell">Wed</div>
            <div className="dashboard__cell">Thu</div>
            <div className="dashboard__cell">Fri</div>
            <div className="dashboard__cell">Sat</div>
            <div className="dashboard__cell">Sun</div>
          </div>
          {checkComponents}
        </div>
      </div>

      <button className="dashboard__add" onClick={() => {}}>
        Add habit
      </button>
    </div>
  );
};

export default Dashboard;
