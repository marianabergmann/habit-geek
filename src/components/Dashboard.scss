.dashboard__container {
  display: flex;
}

.dashboard__habits {
  width: 100px;
}

.dashboard__habits,
.dashboard__days {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
}

.dashboard__row {
  height: 30px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.dashboard__cell {
  width: 35px;
  text-align: center;
}

.dashboard__add {
  margin: 10px;
}
