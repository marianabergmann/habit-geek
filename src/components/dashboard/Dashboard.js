import React from "react";

import SaveHabit from "../habits/SaveHabit";
import DashboardView from "./DashboardView";

const Dashboard = ({
  habits,
  name,
  createHabit,
  isCreatingHabit,
  toggleIsCreatingHabit,
  toggleDayHabit
}) => {
  return isCreatingHabit ? (
    <SaveHabit
      createHabit={createHabit}
      toggleIsCreatingHabit={toggleIsCreatingHabit}
    />
  ) : (
    <DashboardView
      username={name}
      habits={habits}
      toggleIsCreatingHabit={toggleIsCreatingHabit}
      toggleDayHabit={toggleDayHabit}
    />
  );
};

export default Dashboard;
