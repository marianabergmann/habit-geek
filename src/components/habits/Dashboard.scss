.dashboard {
  width: 100%;
  max-width: 600px;
  padding: 15px;
  margin: auto;

  &__title {
    text-align: center;
  }

  &__table {
    display: flex;
    flex-direction: column;
    justify-content: space-around;
  }

  &__row {
    display: flex;
  }

  &__cell-habit-container {
    width: 140px;
  }

  &__cell-frequency-container {
    flex-grow: 1;
    display: flex;
    justify-content: space-around;
  }

  &__cell {
    width: 60px;
    text-align: center;
  }
}

@media (max-width: 576px) {
  .dashboard {
    &__table {
      max-width: 400px;
      margin: auto;
    }

    &__row {
      flex-direction: column;
      margin-bottom: 5px;
    }

    &__cell-habit-container {
      width: 100%;
      text-align: center;
    }

    &__cell {
      font-size: 1rem;
    }
  }
}
