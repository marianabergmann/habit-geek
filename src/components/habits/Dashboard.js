import React, { Fragment } from "react";

import "./Dashboard.scss";
import DashboardRows from "./DashboardRows";

const Dashboard = ({
  username,
  habits,
  toggleIsCreatingHabit,
  toggleDayHabit
}) => {
  return (
    <div className="dashboard">
      <h2 className="mb-3 dashboard__title">
        What have you done this week{username ? " " + username : ""}?
      </h2>

      {habits.length > 0 ? (
        <Fragment>
          <div className="mb-3 dashboard__table">
            <div className="dashboard__row">
              <div className="dashboard__cell-habit-container"></div>
              <div className="dashboard__cell-frequency-container">
                <h4 className="dashboard__cell">Mon</h4>
                <h4 className="dashboard__cell">Tue</h4>
                <h4 className="dashboard__cell">Wed</h4>
                <h4 className="dashboard__cell">Thu</h4>
                <h4 className="dashboard__cell">Fri</h4>
                <h4 className="dashboard__cell">Sat</h4>
                <h4 className="dashboard__cell">Sun</h4>
              </div>
            </div>
            {
              <DashboardRows
                habits={habits}
                toggleDayHabit={toggleDayHabit}
              ></DashboardRows>
            }
          </div>
          <hr />
        </Fragment>
      ) : null}

      <div className="d-flex justify-content-center">
        <button
          className="btn btn-primary btn-lg dashboard__add"
          onClick={toggleIsCreatingHabit}
        >
          I'll add a new habit!
        </button>
      </div>
    </div>
  );
};

export default Dashboard;
