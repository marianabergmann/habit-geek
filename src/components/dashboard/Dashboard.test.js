import React from "react";
import { shallow } from "enzyme";
import Dashboard from "./Dashboard";

describe("Dashboard component", () => {
  it("Renders Dashboard list", () => {
    const dashboard = shallow(<Dashboard isCreatingHabit={false} />);
    expect(dashboard).toMatchSnapshot();
  });

  it("Renders new Habit to create a new one", () => {
    const dashboard = shallow(<Dashboard isCreatingHabit={true} />);
    expect(dashboard).toMatchSnapshot();
  });
});
