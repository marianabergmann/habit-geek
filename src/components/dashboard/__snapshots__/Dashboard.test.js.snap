// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component Renders Dashboard list 1`] = `<DashboardView />`;

exports[`Dashboard component Renders new Habit to create a new one 1`] = `<SaveHabit />`;
